python/tests/command/call_graph_test.py

```diff
@@ -0,0 +1,51 @@
+import os
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from cog.command.call_graph import analyze_python_file
+
+
+def test_call_graph():
+    with tempfile.TemporaryDirectory() as tmpdir:
+        filepath = os.path.join(tmpdir, "predict.py")
+        with open(filepath, "w", encoding="utf8") as handle:
+            handle.write("""from cog import Path, Input
+from cog.ext.pipelines import include
+
+flux_schnell = include("black-forest-labs/flux-schnell")
+
+def run(
+    prompt: str = Input(description="Describe the image to generate"),
+    seed: int = Input(description="A seed", default=0)
+) -> Path:
+    output_url = flux_schnell(prompt=prompt, seed=seed)[0]
+    return output_url
+""")
+        includes = analyze_python_file(Path(filepath))
+        assert includes == ["black-forest-labs/flux-schnell"]
+
+
+def test_call_graph_with_dynamic_string():
+    with tempfile.TemporaryDirectory() as tmpdir:
+        filepath = os.path.join(tmpdir, "predict.py")
+        with open(filepath, "w", encoding="utf8") as handle:
+            handle.write("""from cog import Path, Input
+from cog.ext.pipelines import include
+
+for i in range(50):
+    flux_schnell = include(f"black-forest-labs/flux-schnell-{i}")
+
+def run(
+    prompt: str = Input(description="Describe the image to generate"),
+    seed: int = Input(description="A seed", default=0)
+) -> Path:
+    output_url = flux_schnell(prompt=prompt, seed=seed)[0]
+    return output_url
+""")
+        with pytest.raises(ValueError) as excinfo:
+            analyze_python_file(Path(filepath))
+        assert str(excinfo.value).endswith(
+            "Unresolvable argument at line 5: Not a string literal"
+        )
```

Comment:
Nice, the other constraint currently is that we only support calling `include()` at the top level of a module. Dynamically calling it within a predictor for example is no good, so we should either, a) limit the parser to only walk the module scope or b) walk the entire file and error if a call is found outside of the module scope.

We could also do this as a follow up if needed.